sklearn/tree/tests/test_tree.py

```diff
@@ -2524,3 +2524,17 @@ def test_n_features_deprecated(Tree):
 
     with pytest.warns(FutureWarning, match=depr_msg):
         Tree().fit(X, y).n_features_
+
+
+def test_tree_split_with_identical_feature():
```

Comment:
I would parametrize the test with several `random_state` and try at least 10 seeds.